framework_tvb/tvb/core/services/simulator_service.py

```diff
@@ -155,19 +156,24 @@ def async_launch_and_prepare_pse(self, burst_config, user, project, simulator_al
             operation_group = burst_config.operation_group
             metric_operation_group = burst_config.metric_operation_group
             operations = []
-            range_param2_values = []
+            range_param2_values = [None]
             if range_param2:
                 range_param2_values = range_param2.get_range_values()
             first_simulator = None
-            for param1_value in range_param1.get_range_values():
-                for param2_value in range_param2_values:
+
+            for param2_value in range_param2_values:
+                for param1_value in range_param1.get_range_values():
```

Comment:
Why was this switch of the loops necessary?
Theoretically it gives the same set of iterations, just in a different order, and it is not very easy to digest when reading the code.

Reply:
Yeah, I realized it is not necessary... The important change there was that in case of a single parameter to make range_param2_values be a list = [None] so it can still enter in the fors and compute the array of operation. Original order can stay.

Reply:
And about the PSE_RANGE_PARAM_1 and 2 (I dunno why it doesn't let me directly reply to that comment), the thing is that those range params are needed in launch_pse, but they don't get it in the data, because that final fragment doesn't have them and range params are also not part of session_stored_simulator. So how can I do it differently without having to store them in the session?

Reply:
I would expect that we use now forms everywhere.
Indeed, these can not stay on a seesion_stored_sim, but there should be a form displaying the ranges. Isn't it?

Reply:
There is. SimulatorPSEConfigurationFragment is for choosing the params, and SimulatorPSEParamRangeFragment for choosing the ranges. But the params are needed not just in their respective setting method, but in launch_pse as well.